File: 55.jump-game.py
class Solution:
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        max_pos = 0
        i = 0
        while i < len(nums) and i <= max_pos:
            max_pos = max(max_pos, i + nums[i])
            i += 1
        return max_pos >= len(nums) - 1

File: 55.jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int max_pos = 0;
        for (int i = 0; i < nums.size() && i <= max_pos; i++) {
            max_pos = max(max_pos, i + nums[i]);
        }
        return max_pos >= nums.size() - 1;
    }
};